vim25/soap/client.go

```diff
@@ -503,10 +504,13 @@ func (c *Client) Do(ctx context.Context, req *http.Request, f func(*http.Respons
 		defer d.done()
 	}
 
-	if c.UserAgent != "" {
-		req.Header.Set(`User-Agent`, c.UserAgent)
+	// use default
+	if c.UserAgent == "" {
+		c.UserAgent = fmt.Sprintf("%s/%s", version.SDKName, version.SDKVersion)
```

Comment:
Alternative, this could be a custom (exported) `UserAgentFn` which produces the string. I went for the simple (hard-coded) approach here.

Reply:
Let's leave it as hard-coded default, since caller can already set `Client.UserAgent` to override. Also with it hard-coded can use a const like above comment, `internal.UserAgent` instead of Sprintf.

Reply:
Can't use const for the `UserAgent` because it's generated at compile time using `fmt.Sprintf` and `runtime.X` variables.